src/sparse/impl/KokkosSparse_twostage_gauss_seidel_impl.hpp

```diff
@@ -742,7 +866,7 @@ namespace KokkosSparse{
                   y_value_array_type localB, // in
                   bool init_zero_x_vector = false,
                   int numIter = 1,
-                  scalar_t omega = Kokkos::Details::ArithTraits<scalar_t>::one(),
+                  scalar_t omega = ST::one(),
```

Comment:
Could you add a runtime check that the ``omega`` passed in here matches ``gsHandle->getOuterDampFactor()``? I think it's fine to do the triangular scaling by ``omega`` during setup rather than apply, but checking for that might prevent confusing behavior. I'm worried the user might thing passing it to apply() is all they need to do, since that's how MTGS works.

Reply:
Hi, @brian-kelley. The damping factor is currently used only for two-stage iteration (I had the sptrsv option mainly for debugging before the damping factor is integrated).

For using sptrsv, I am explicitly extracting the triangular part during the setup, but I don't have omega till apply, so I am assuming to be one. Then sptrsv is called for apply, so omega does not do anything. I can check omega=one at runtime during the apply with sptrsv, but what should I do if it is not omega?

Reply:
@iyamazaki I see. If this path is mainly for debugging, I think it's fine to just do
```if(omega != 1.0) throw invalid_argument(...)```.

But if it's easy, you could make this path support arbitrary omega by adding extra BLAS calls.

Reply:
I added throw invalid_argument, thank you @brian-kelley.

Reply:
Hi, @brian-kelley, again.. Apparently, we test this option with omega=0.9.. I reset omega to be 1.0 for the unit-test to pass. I hope this is okay.

Reply:
That's fine.